MapboxVision/src/main/java/com/mapbox/vision/visionevents/events/roaddescription/RoadDescription.kt

```diff
@@ -25,70 +24,85 @@ data class RoadDescription(
+                                        worldPoints = emptyList()// TODO fill lane points
+                                )
+                        )
+                )
             }
 
-            val currentTrackLeftLineWorldPoints = emptyList<WorldCoordinate>()
-            val currentTrackLeftMarking = Marking(MarkingType.SOLID, currentTrackLeftLineWorldPoints)
-
-            val currentTrackRightLineWorldPoints = emptyList<WorldCoordinate>()
-            val currentTrackRightMarking = Marking(MarkingType.SOLID, currentTrackRightLineWorldPoints)
-
-            val currentLine = Line(0.0, Direction.FORWARD, currentTrackLeftMarking, currentTrackRightMarking)
-            lines.add(currentLine)
-
-
-            // Fill right lines
-            for (i in 0 until visibleRightLanes) {
-
-                val isRightBorder = (i == 0)
-
-                val leftMarkingType = if (isRightBorder && seeRightBorder) {
-                    MarkingType.CURB
-                } else {
-                    MarkingType.UNKNOWN
+            for (laneIndex in (0..sameDirectionVisibleLanes)) {
+                val leftMarkingType = when {
+                    laneIndex != 0 -> MarkingType.DASHES
+                    visibleReverseLanes != 0 -> MarkingType.DOUBLE_SOLID
+                    seeLeftBorder -> MarkingType.CURB
+                    else -> MarkingType.UNKNOWN
                 }
 
-                val leftLineWorldPoints = emptyList<WorldCoordinate>()
-                val leftMarking = Marking(leftMarkingType, leftLineWorldPoints)
-
-                val rightLineWorldPoints = emptyList<WorldCoordinate>()
-                val rightMarking = Marking(MarkingType.SOLID, rightLineWorldPoints)
-
-
-                val line = Line(0.0, Direction.FORWARD, leftMarking, rightMarking)
-
-                lines.add(line)
+                val rightMarkingType = when {
+                    laneIndex != sameDirectionVisibleLanes - 1 -> MarkingType.DASHES
+                    seeRightBorder -> MarkingType.CURB
+                    else -> MarkingType.UNKNOWN
+                }
 
+                lines.add(
+                        Line(
+                                width = width,
+                                direction = Direction.FORWARD,
+                                leftMarking = Marking(
+                                        leftMarkingType,
+                                        worldPoints = emptyList()// TODO fill lane points
+                                ),
+                                rightMarking = Marking(
+                                        rightMarkingType,
+                                        worldPoints = emptyList()// TODO fill lane points
+                                )
+                        )
+                )
             }
 
-            return RoadDescription(roadDescriptionDataBuffer.roadDescriptionIdentifier, lines, visibleLeftLanes, egoOffset)
+            return RoadDescription(
```

Comment:
Why did you change it. Do you need roadDescription object?

Reply:
Yeah I needed width in previous distance to car version - but now use updated world description so dont need roaddescription. However - I've aligned it with iOS implementation already.